src/tools/ranges/TransformView/Iterator.hpp

```diff
@@ -0,0 +1,182 @@
+
+    ++this->iter_;
+    return *this;
+  }
+
+  constexpr Iterator operator++( int ) {
+
+    return Iterator( this->iter_++, transform_ );
+  }
+
+  constexpr Iterator& operator--( void ) {
+
+    static_assert(
+      concepts::IsBidirectionalIterator< Iterator >::value == true,
+      "the at() method can only be made available for bidirectional iterators" );
+
+    --this->iter_;
+    return *this;
+  }
+
+  constexpr Iterator operator--( int ) {
+
+    static_assert(
+      concepts::IsBidirectionalIterator< Iterator >::value == true,
+      "the at() method can only be made available for bidirectional iterators" );
+
+    return Iterator( this->iter_--, this->transform_ );
+  }
+
+  friend constexpr Iterator operator+( difference_type i, Iterator iter ) {
+
+    static_assert(
+      concepts::IsRandomAccessIterator< Iterator >::value == true,
+      "the at() method can only be made available for random access iterators" );
+
+    return Iterator( iter.iter_ + i, iter.transform_ );
+  }
+
+  friend constexpr Iterator operator+( Iterator iter, difference_type i ) {
+
+    static_assert(
+      concepts::IsRandomAccessIterator< Iterator >::value == true,
+      "the at() method can only be made available for random access iterators" );
+
+    return Iterator( iter.iter_ + i, iter.transform_ );
+  }
+
+  friend constexpr Iterator operator-( Iterator iter, difference_type i ) {
+
+    static_assert(
+      concepts::IsRandomAccessIterator< Iterator >::value == true,
+      "the at() method can only be made available for random access iterators" );
+
+    return Iterator( iter.iter_ - i, iter.transform_ );
+  }
+
+  friend constexpr difference_type operator-( const Iterator& left,
+                                              const Iterator& right ) {
+
+    static_assert(
+      concepts::IsRandomAccessIterator< Iterator >::value == true,
+      "the at() method can only be made available for random access iterators" );
+
+    return left.iter_ - right.iter_;
+  }
+
+  friend constexpr bool operator==( const Iterator& left,
+                                    const Iterator& right ) {
+
+    return left.iter_ == right.iter_;
+  };
+
+  friend constexpr bool operator!=( const Iterator& left,
+                                    const Iterator& right ) {
+
+    return !( left == right );
+  };
+
+  friend constexpr bool operator<( const Iterator& left,
+                                   const Iterator& right) {
+
+    static_assert(
+      concepts::IsRandomAccessIterator< Iterator >::value == true,
+      "the at() method can only be made available for random access iterators" );
+
+    return left.iter_ < right.iter_;
+  }
+
+  friend constexpr bool operator>( const Iterator& left,
+                                   const Iterator& right ) {
+
+    static_assert(
+      concepts::IsRandomAccessIterator< Iterator >::value == true,
+      "the at() method can only be made available for random access iterators" );
```

Comment:
Copy-paste errors on all of these static-asserts - they all reference the 'at()' method instead of the appropriate method.

Also, I may not have realized that doing this instead of SFINAE would require static asserts on all of the methods.  Sorry about that.

Reply:
Ah, I see you fixed it in the next MR

Reply:
Indeed, I fixed it later.

I don;t mind all those static asserts. It;s more readable compared to sfinae